File: lib/utils/formatters.dart
import 'package:intl/intl.dart';
import '../models/currency.dart';

String formatCurrency(double amount, String currencyCode, List<Currency> currencies) {
  final currency = currencies.firstWhere(
    (c) => c.code == currencyCode,
    orElse: () => currencies.isNotEmpty ? currencies.first : Currency(
      id: '',
      code: 'USD',
      symbol: '\$',
      name: 'Dólar',
    ),
  );

  final formatter = NumberFormat.currency(
    symbol: currency.symbol,
    decimalDigits: 2,
  );

  return formatter.format(amount);
}

String formatDate(DateTime date) {
  return DateFormat('yyyy-MM-dd').format(date);
}

String formatDateShort(DateTime date) {
  return DateFormat('MMM dd', 'es').format(date);
}

File: src/utils/formatters.ts
import { format, parseISO } from 'date-fns';
import { Currency } from '../models/Currency';

export const formatCurrency = (
  amount: number,
  currencyCode: string,
  currencies: Currency[]
): string => {
  const currency = currencies.find(c => c.code === currencyCode) || currencies[0] || {
    id: '',
    code: 'USD',
    symbol: '$',
    name: 'Dólar',
  };

  return `${currency.symbol}${amount.toFixed(2).replace(/\B(?=(\d{3})+(?!\d))/g, ',')}`;
};

export const formatDate = (date: Date | string): string => {
  const dateObj = typeof date === 'string' ? parseISO(date) : date;
  return format(dateObj, 'yyyy-MM-dd');
};

export const formatDateShort = (date: Date | string): string => {
  const dateObj = typeof date === 'string' ? parseISO(date) : date;
  return format(dateObj, 'MMM dd');
};

